scipy/integrate/_quadrature.py

```diff
@@ -312,9 +313,19 @@ def vfunc(x):
     return vfunc
 
 
+@_deprecated("`scipy.integrate.quadrature` is deprecated as of SciPy 1.12.0"
+             "and will be removed in SciPy 1.14.0. Please use "
+             "`scipy.integrate.quad` for new code or "
+             "`scipy.integrate.gauss_quad` for legacy code.")
 def quadrature(func, a, b, args=(), tol=1.49e-8, rtol=1.49e-8, maxiter=50,
                vec_func=True, miniter=1):
     """
+    .. deprecated:: 1.12.0
+
+        This function is deprecated and will be removed in SciPy 1.14.0.
+        Please use `scipy.integrate.quad` for new code or
+        `scipy.integrate.gauss_quad` for legacy code.
```

Comment:
This LGTM, but please suggest alternative wording if desired.

![image](https://github.com/scipy/scipy/assets/6570539/9322fd32-f95b-447a-8059-4937ace51f14)